### PR TITLE
Added floor function to numpy frontend

### PR DESCRIPTION
**Pull request by RakshitKumar04**
Close https://github.com/unifyai/ivy/pull/12948

### CODE REVIEW

**Comment by RakshitKumar04** [comment]
Close #12948

**Comment by RakshitKumar04** [comment]
@hirwa-nshuti Created the new PR and added the floor function, please review it.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_rounding.py [hunk 1]
```diff
@@ -76,6 +76,50 @@ def test_numpy_fix(
     )
 
 
+#floor
+@handle_frontend_test(
+    fn_tree="ivy.floor",
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_rounding.py:81]: fn_tree should be `numpy.floor`

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @RakshitKumar04 

Thanks for contributing to ivy😊

Here are general comments for this implementation:

* You need to import this private function in the init file of the numpy frontend: https://github.com/unifyai/ivy/blob/6c29acd94482e341148a629df523cf966d02aa4b/ivy/functional/frontends/numpy/__init__.py#L518

* You'll need to initialize ufunc object here in the same init file: https://github.com/unifyai/ivy/blob/6c29acd94482e341148a629df523cf966d02aa4b/ivy/functional/frontends/numpy/__init__.py#L676

You'll then fix the test issues refer to the comments below
Thanks

**Comment by RakshitKumar04** [comment]
Hey, @hirwa-nshuti fixed the testing issues, please review.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_rounding.py (hunk 1 above)
Comment (RakshitKumar04) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_rounding.py]: Resolved the conflict

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @RakshitKumar04 

The tests are failing due to wrong implementation of the unit test 🙂
Would you follow the unit test of `ceil` function above to understand anover view as the tests will be somehow alike😊

Thanks

**Comment by RakshitKumar04** [comment]
> Hey @RakshitKumar04
> 
> The tests are failing due to wrong implementation of the unit test slightly_smiling_face Would you follow the unit test of `ceil` function above to understand anover view as the tests will be somehow alikeblush
> 
> Thanks

Hey, did the changes you suggested, Please review

**Review by hirwa-nshuti — APPROVED** [review]
Looks good!

Going to merge
Thanks for contributing😊

**Comment by RakshitKumar04** [comment]
> Looks good!
> 
> Going to merge Thanks for contributingblush

Thank you @hirwa-nshuti for helping me contribute to ivy.

**Pull request merged**